# SQL Part 2 Studio

### Let's practice your SQL quering skills!  For each question, work along in the notebook adding your query and answering the questions. 

### **DataSet**

### <span style="font-size: 14px; font-family: -apple-system, BlinkMacSystemFont, sans-serif;">We will be using the Goodbooks-10k dataset from the exercises in the prep work.&nbsp; Feel free reference your notebook.</span>

- The dataset can be found here: [goodbooks-10k](https://www.kaggle.com/zygmunt/goodbooks-10k)
- You can access the **BooksDB** in the Launchcode server.

### **Business Issue**

You are an employee at a small independet bookstore and you have been tasked to increase sales.  You decide to use the BooksDB to find bookss and themes to highlight in fun promotions throughout each month.  We will be starting with March and then you will have a chance to explore another month of your choosing.  We want to come up with a list of at least 3 events/ promotions to run each month.  If you are looking for ideas to include in your promotions:

- [https://www.calendarr.com/united-states/calendar-2022/](https://www.calendarr.com/united-states/calendar-2022/)
- [https://www.holidayinsights.com/moreholidays/](https://www.holidayinsights.com/moreholidays/)
    - Click on a month and it will take you to a more detailed page

## Part 1:  March - Womans History Month, Dr Seuss Day(3/2), National Pie Day3/14, St Patricks Day (3/17), James Patterson(3/22)

Event 1: Womans History Month

Goal: Highlight Popular Womans Authors and Issues

In [4]:
-- SOLUTION
-- Query to look at tags to see which ones may be helpful
SELECT TOP 1000 * 
   FROM BooksDB.dbo.tags
   WHERE tag_name LIKE '%woman%'
    

(24 rows affected)

Total execution time: 00:00:00.098

tag_id tag_name 937 21-books-every-woman-should-read 1386 a-woman-after-god-s-own-heart 1387 a-woman-after-gods-own-heart 4006 batwoman 4455 biblical-womanhood 4729 black-ops-woman-team-cia-sean-case 5132 books-every-woman-should-read 7194 christian-woman 10986 every-woman-should-read 19038 man-woman 19757 memorable-woman-characters 22269 older-man-younger-woman 22270 older-woman 22271 older-woman-younger-man 23671 pioneer-woman 23891 policewoman-potential-victim 32679 woman-author 32680 woman-s-fiction 32681 woman-s-murder-club 32682 woman-trans-agender-authors 32683 womanhood 32684 womans-literature 32685 womans-murder-club 32740 wonder-woman

Decide I want to explore the following tags:

line 17 tag\_id is woman-author, 32679

line 20 tag\_id is woman-trans-agender-author, 32682

line 21 tag\_id is womanhood, 32683

  

### Join Tables to return authors, titles with tags selected, ratings.

Decide I want to return the top 10 rated books with each tag to highlight

In [ ]:
SELECT TOP 10(title)
FROM BooksDB.dbo.books
WHERE 

Commands completed successfully.

Total execution time: 00:00:00

# Part 2: Choose Another Month

Choose another month and plan at least 3 events / promotions and answer the following questions.

Which month did you choose?  

What 3 events / promotions are you highlighting?

> 1.

> 2.

> 3.

# Part 3: Summarize your Work

For each event write at least one query to support your choice and record you thoughts as to why you used the paticlular query.

In [ ]:
-- Event 1

In [ ]:
-- Event 2

In [ ]:
-- Event 3